target/linux/lantiq/base-files/etc/uci-defaults/02_migrate_xdsl_iface

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+. /lib/functions.sh
+. /lib/functions/lantiq.sh
+
+WAN_IFACE_CHANGED=0
+
+if lantiq_is_vdsl_system; then
+	config_load network
+	config_get iface wan ifname
+
+	case $iface in
+		nas0*)
+			iface=${iface/nas0/dsl0}
+			WAN_IFACE_CHANGED=1
+			;;
+		ptm0*)
+			iface=${iface/ptm0/dsl0}
+			WAN_IFACE_CHANGED=1
+			;;
+	esac
+
+	uci set network.wan.ifname=$iface
```

Comment:
Shouldn't it be enough to add a 'uci set network.wan_dev.name=dsl0' here ?

Reply:
Nope. It would drop the vlan tag again. For example ptm0.7 need to be dsl0.7 to make t-com connections working.

Reply:
Bummer, you are talking about wan_dev.

Yes, dsl0 should be enough to match the default config. But you never know what people have in their config.

They might have changed it to a vlan interface (not sure if this is even supported) or they are using this section for a complete different purpose. To be on the safe side, it should checked if the name matches the expections and we should expect vlan interfaces here as well.

Reply:
OK, then I will insert the complete check again here.